mne_rsa/viz.py

```diff
@@ -67,3 +69,167 @@ def plot_dsms(dsms, names=None, items=None, n_rows=1, cmap='viridis',
+        Layout objects containing sensor layout info.
+        The default (``None``) will figure out layout based on info.
+    fig: matplotlib.pyplot.Figure | None
+        Figure object on which DSMs on 2D MEG topography are plotted.
+        The default (``None``) creates a new Figure object.
+    title: str | None
+        Title of the plot, used only when ``fig=None``.
+        The default (``None``) puts no title in the figure.
+    axis_facecolor: str
+        Face color of the each DSM. Defaults to 'w', white.
+    axis_spinecolor: str
+        Spine color of each DSM. Defaults to 'w', white.
+    fig_facecolor: str
+        Face color of the entire topography. Defaults to 'w', white.
+    figsize: tuple of float
+        Figure size. The first element specify width and the second height.
+        Defaults to (6.4, 4.8).
+    cmap: str
+        Colormap used for plotting DSMs. Defaults to 'viridis'.
+        Check :func:`matplotlib.pyplot.imshow` for details.
+    show: bool
+        Whether to display the generated figure. Defaults to False.
+
+    Returns
+    -------
+    fig: matplotlib.pyplot.Figure
+        Figure object in which DSMs are plotted on 2D MEG topography.
+    """
+    on_pick = partial(_click_func, dsms=dsms, cmap=cmap)
+
+    if fig is None:
+        fig = plt.figure(figsize=figsize)
+        if title is not None:
+            fig.suptitle(title, x=0.98, horizontalalignment='right')
+    else:
+        fig = plt.figure(fig.number)
+
+    my_topo_plot = _iter_topography(info=info, layout=layout, on_pick=on_pick,
+                                    fig=fig, axis_facecolor=axis_facecolor,
+                                    axis_spinecolor=axis_spinecolor,
+                                    fig_facecolor=fig_facecolor,
+                                    unified=False)
+
+    for i, (ax, _) in enumerate(my_topo_plot):
+        dsms_i = dsms[i]
+        dsms_i = distance.squareform(dsms_i)
+        ax.imshow(dsms_i, cmap=cmap)
+
+    if show:
+        fig.show()
+
+    return fig
+
+
+def plot_dsms_topo(dsms, info, time_windows=None, layout=None, figs=None,
+                    axis_facecolor='w', axis_spinecolor='w',
+                    fig_facecolor='w', figsize=(6.4, 4.8), cmap='viridis',
+                    show=True):
+    """ Plot DSMs on 2D MEG topographies for multiple time windows
+
+    Parameters
+    ----------
+    dsms: ndarray | numpy.memmap, shape (n_sensors, n_timepoints, n_dsm_datapoints)
+        DSMs of MEG recordings; one DSM for each sensor and time point.
+    info: mne.io.meas_info.Info
+        Info object that contains meta data of MEG recordings.
+    time_windows: list of [int, int] | None
+        List of time windows for each of which
+            the average DSM is calculated and plotted. By default, the average of DSMs of
+            all the time points is plotted. Start of time window is inclusive, while end is
+            exclusive.
```

Comment:
incorrect indentation.

But more importantly, from a user's perspective, this interface for specifying time is clumsy. It makes the simplest case, plotting the DSMs at a single time point, overly complicated.

Let's see, here are some use cases the user may have:

 1. Plot the DSMs at a certain time
 1. Plot the DSMs at multiple time points, no averaging
 1. Plot the average DSM given a single time window 
 1. Plot multiple average DSMs given multiple time windows
 1. Plot the average DSM, averaged across all available time points

The first use case is the most common I would say, so this should be super easy. The last use case can be a good default when no time is given. Use case 2 is easy for a user to do with a for loop. Use case 3 is a bit trickier so there it would be beneficial for the `plot_dsms_topo` function to provide functionality. Use case 5 is again trivial with a for loop if use case 3 is easy to do.

Looking at the list above, I would say, scrap the functionality to generate multiple figures from this function. Instead, allow time to be either `None` to average across all time, a single int to plot the DSMs at a single time point or a tuple (or other sequence of length 2) of integers to plot the average DSMs in the given time window.